drivers/pinmux/pinmux_stm32.c

```diff
@@ -104,20 +104,12 @@ static int stm32_pin_configure(uint32_t pin, uint32_t func, uint32_t altf)
 		return -ENODEV;
 	}
 
-#ifdef CONFIG_PM_DEVICE_RUNTIME
-	ret = pm_device_runtime_get(port_device);
-	if (ret != 0) {
+	ret = gpio_stm32_configure(port_device, STM32_PIN(pin), func, altf);
+	if (ret < 0) {
```

Comment:
Same is required on pinctrl_stm32.c

Reply:
adjusted too

Reply:
Can you return ret here instead and avoid the added branch?

Reply:
done